src/js/angular/import/controllers.js

```diff
@@ -1030,41 +1030,44 @@ importViewModule.controller('TabCtrl', ['$scope', '$location', 'ImportViewStorag
             ImportViewStorageService.setHelpVisibility(true);
             shouldResetHelpVisibility = false;
         }
-        const viewPersistance = ImportViewStorageService.getImportViewSettings();
-        let isVisible = viewPersistance.isHelpVisible;
-        if (files.length === 0 && viewPersistance.isHelpVisible) {
+        const viewPersistence = ImportViewStorageService.getImportViewSettings();
+        let isVisible = viewPersistence.isHelpVisible;
+        if (files.length === 0 && viewPersistence.isHelpVisible) {
             isVisible = true;
-        } else if (files.length === 0 && !viewPersistance.isHelpVisible) {
+        } else if (files.length === 0 && !viewPersistence.isHelpVisible) {
             isVisible = false;
-        } else if (viewPersistance.isHelpVisible) {
+        } else if (viewPersistence.isHelpVisible) {
             isVisible = true;
-        } else if (!viewPersistance.isHelpVisible) {
+        } else if (!viewPersistence.isHelpVisible) {
             isVisible = false;
         }
         ImportViewStorageService.setHelpVisibility(isVisible);
         $scope.isHelpVisible = isVisible;
     };
 
     // =========================
-    // Initialization
+    // Watchers and event handlers
     // =========================
 
-    const init = function () {
-        ImportContextService.onFilesUpdated(onFilesUpdated);
-
-        if ($scope.viewType !== 'user' && $scope.viewType !== 'server') {
-            $scope.viewType = 'user';
-        }
-
-        if ($scope.viewType === 'user') {
+    const subscriptions = [];
+    subscriptions.push(ImportContextService.onActiveTabIdUpdated((newActiveTabId) => {
+        $scope.activeTabId = newActiveTabId;
+        $location.hash($scope.activeTabId);
+        if (TABS.USER === $scope.activeTabId) {
             $scope.templateUrl = 'js/angular/import/templates/uploadInfo.html';
         } else {
             $scope.templateUrl = 'js/angular/import/templates/importInfo.html';
         }
+    }));
 
-        $scope.openTab($scope.viewType || 'user');
-    };
-    init();
+    subscriptions.push(ImportContextService.onFilesUpdated(onFilesUpdated));
+
+    const removeAllListeners = () => subscriptions.forEach((subscription) => subscription());
+
+    $scope.$on('$destroy', removeAllListeners);
+
+    // Updates the active tab id from tha url.
+    ImportContextService.updateActiveTabId($location.hash() || TABS.USER);
```

Comment:
Let's try to keep things tidy. There should not be any expression outside of functions.

Reply:
Done

Reply:
Actually it's not done. See how the other controllers are made. Move all initialization logic inside the init function which you actually removed here I think. Then just call the init.